### DISCUSSION TITLE
Low performance while merging a lot of cells

### DISCUSSION THREAD

**Discussion by planemar** [exceljs/exceljs#2528] [Ideas & Proposals]
Hi guys.
I found that this lib has low performance if you need to merge a lot of cells in big document.
E.g for 10000 rows with 20 columns and with 1500 ranges for merging it takes around 10 mins.
I looked into source code and found that the reason of this is the next code
/lib/stream/xlsx/worksheet-writer.js
```
// check cells aren't already merged
this._merges.forEach(function(merge) {
    if (merge.intersects(dimensions)) {
        throw new Error('Cannot merge already merged cells');
    }
});
```

This checking takes a lot of time. Without this checking merging takes less than 1 sec.
Thanks.

**Comment by Imperat** [comment]
I think that this checking takes a lot of time because of it has O(n^2) difficult of calculus. You merge cells N times and for each merge operation this check takes also N times. 
I think it's possible to write this check using binary search algorithm for using O(nlogn) difficult.
